### ISSUE TITLE
Top navigation in mobile view - items in last expandable nav group not focusable

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-ui-kit#3261]
## Summary of the bug 
When viewing a top navigation with navigation groups on a small / mobile screen, if the last navigation group is expandable, the nav items within that group cannot be tabbed to via the keyboard.

## 🪜 How to reproduce 
1. Go to this [top navigation StackBlitz](https://stackblitz.com/edit/ji6krlvb?file=src%2Fapp.tsx&hideExplorer=1).
2. Tab to the third nav group ("Patterns").
3. Press enter to open the nav group.
4. Try tabbing to the nav items within that group.
5. See that the group collapses and the focus moves to the close button.

This seems to only happen on the last navigation group, and if it is expandable. If the last navigation group does not have the `expandable` prop, everything works fine.

You can click on the child nav items but not tab to them.

## 📸 Screenshots or code 
https://github.com/user-attachments/assets/e3b66944-f8fc-4baf-8d36-42d58dff58c6

## 🧐 Expected behaviour 
The navigation group should stay open and let you tab to the child nav items.

**Comment by GCHQ-Developer-299** [comment]
When working on this ticket, add/modify a storybook example for grouped navigation as per the stackblitz

**Comment by GCHQ-Developer-847** [comment]
Remember to remove the skip from the IcNavigationMenu test called "should expand navigation groups and tab to navigation items within expandable navigation groups"
